### ISSUE TITLE
"v2+" module name

### ISSUE THREAD

**Issue by ldez** [fauna/faunadb-go#97]
## (Problem) Description

for the `v2+` version, the go modules need that the module path to sync with the major version.

https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher

## Steps to reproduce

```go
module github.com/yourusername/project

go 1.12

require (
	github.com/fauna/faunadb-go v2.12.0
)
```

Run `go mod tidy`

```
require github.com/fauna/faunadb-go: version "v2.12.0" invalid: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v2
```

## Expected behaviour

```go
module github.com/yourusername/project

go 1.13

require (
	github.com/fauna/faunadb-go v2.12.0
)
```

## Actual behaviour

To be able to use the faunadb client, we are forced to use hash (pseudo-version) instead a real version

```go
module github.com/yourusername/project

go 1.13

require (
	github.com/fauna/faunadb-go v1.0.1-0.20200506155252-ce35719ad41e
)
```

## Suggested fix

Change the module name and update the package inside this repo.

```go
module github.com/fauna/faunadb-go/v2

go 1.11

require (
	github.com/davecgh/go-spew v1.1.1 // indirect
	github.com/pmezard/go-difflib v1.0.0 // indirect
	github.com/stretchr/testify v1.2.2
)
```

```go
package faunadb_test

import (
	"testing"
	"time"

	f "github.com/fauna/faunadb-go/faunadb/v2"
	"github.com/stretchr/testify/suite"
)

```

**Comment by trevorsibanda** [comment]
Thank you for opening a ticket.

Related to #84 , #77 , #90

**Comment by ldez** [comment]
also as it's a breaking change, as it's recommended by the [go team](https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher), I think you have to create a new major version (v3).

FYI you don't need to create a sub-directory `v2` or `v3`.

**Comment by bithavoc** [comment]
The only walkaround so far has been `go get github.com/fauna/faunadb-go@master`.

**Comment by ldez** [comment]
It's not the only workaround: you can use commit hash and pseudo version.

**Comment by ldez** [comment]
Fixed by #112
